### PR TITLE
ERR: Improve error message for Series.loc.getitem with too many dimensions

### LINKED ISSUE #35349: Non-intuitive behavior when multi-indexing a Series
Most likely that is not a bug but intended behavior but I find it quite confusing.

Consider the following example
```
idx = pd.IndexSlice
s = pd.Series(index=pd.MultiIndex.from_tuples([('A', '0'), ('A', '1'), ('B', '0')]), 
              data=[21, 22, 23])
s.loc[idx['A', :], :]
```
which gives
```
A  0    21
   1    22
B  0    23
dtype: int64
```
while intuitively, one would expect
```
A  0    21
   1    22
dtype: int64
```
I would expect the latter because that's what you would get with a `pd.DataFrame` where the second `:` is referring to the column index. Not paying attention that the object is a series, gives you a completely different result than if you would have a data frame.

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #35349
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

tuple in tuple means we are indexing index and columns

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. can you also test setitem. if its not trivial, can do in a separate PR/issue.

**Comment by phofl** [comment]
Good point. This is pretty straight forward

**Review by jreback — APPROVED** [review]
lgtm can you merge master and ping on green

File: doc/source/whatsnew/v1.3.0.rst [hunk 1]
```diff
@@ -53,6 +53,7 @@ Other enhancements
 - :meth:`DataFrame.apply` can now accept non-callable DataFrame properties as strings, e.g. ``df.apply("size")``, which was already the case for :meth:`Series.apply` (:issue:`39116`)
 - :meth:`Series.apply` can now accept list-like or dictionary-like arguments that aren't lists or dictionaries, e.g. ``ser.apply(np.array(["sum", "mean"]))``, which was already the case for :meth:`DataFrame.apply` (:issue:`39140`)
 - :meth:`.Styler.set_tooltips` allows on hover tooltips to be added to styled HTML dataframes.
+- :meth:`Series.loc.__getitem__` and :meth:`Series.loc.__setitem__` with :class:`MultiIndex` now raising helpful error message when indexer has too many dimensions (:issue:`35349`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.3.0.rst:56]: could also put this in bug fixes indexing

Comment (phofl) [reply doc/source/whatsnew/v1.3.0.rst:56]: Could move it if you prefer.

Since this was already raising but with a cryptic error message I put it there

**Comment by jreback** [comment]
thanks @phofl

**Comment by jbrockmendel** [comment]
@phofl I think this introduces a problem if the MultiIndex has a level that includes tuples e.g. (xref #27591)

```
lev1 = ["a", "b", "c"]
lev2 = [(0, 1), (1, 0)]
lev3 = [0, 1]
cols = pd.MultiIndex.from_product([lev1, lev2, lev3], names=["x", "y", "z"])
df = pd.DataFrame(index=range(5), columns=cols)
[test for DataFrame behavior (also broken xref #27591)

ser = df.iloc[0]
result2 = ser.loc[lev1[0], lev2[0], lev3[0]]  # <-- raises incorrectly
expected2 = ser.iloc[:1]
tm.assert_series_equal(result2, expected2)
```

**Pull request merged**